filter.go

```diff
@@ -227,10 +227,14 @@ func (m Middleware) ServeHTTP(w http.ResponseWriter, r *http.Request, next caddy
 		replaced := m.compiledSearchRegex.ReplaceAllFunc(buf.Bytes(), func(input []byte) []byte {
 			return m.replacer(input, repl) // forward the replacement processing
 		})
+
+		csr.Header().Set("Content-Length", strconv.Itoa(len(replaced)))
```

Comment:
Can you add an `if` check here to only set `Content-Length` when it already exists in `csr.Header()`? Need this because in some cases Content-Length is forbidden in headers (https://datatracker.ietf.org/doc/html/rfc7230#section-3.3.2)

Reply:
done see 9e8ac1a